From: "Dr. Tobias Quathamer" <toddy@debian.org>
Date: Fri, 6 Feb 2026 14:06:33 +0100
Subject: Replace localhostCert and localhostKey.

The previous certificate expired on 2026-03-18 19:27:54 UTC.
A new certificate can be generated by running this command
in the directory src/crypto/tls:

$ go run generate_cert.go --rsa-bits 2048 --host 127.0.0.1,::1,example.com --ca --start-date "Jan 1 00:00:00 1970" --duration=1000000h

This patch adds a newly generated certificate which expires
on 2084-01-29 16:00:00 UTC.

See Debian BTS #1127117 (filed for golang-1.25)
---
 src/net/smtp/smtp_test.go | 87 ++++++++++++++++++++++++-----------------------
 1 file changed, 44 insertions(+), 43 deletions(-)

diff --git a/src/net/smtp/smtp_test.go b/src/net/smtp/smtp_test.go
index 427ed0f..c4aeae6 100644
--- a/src/net/smtp/smtp_test.go
+++ b/src/net/smtp/smtp_test.go
@@ -1140,54 +1140,55 @@ func sendMail(hostPort string) error {
 //		--ca --start-date "Jan 1 00:00:00 1970" --duration=1000000h
 var localhostCert = []byte(`
 -----BEGIN CERTIFICATE-----
-MIIDFDCCAfygAwIBAgIRAPV4ktbcY/mn0oRRjnGAGJgwDQYJKoZIhvcNAQELBQAw
-EjEQMA4GA1UEChMHQWNtZSBDbzAeFw0yNTAzMTgxOTI3NTRaFw0yNjAzMTgxOTI3
-NTRaMBIxEDAOBgNVBAoTB0FjbWUgQ28wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw
-ggEKAoIBAQDbsEfk1bK7ozwZlcQM8rBUikC4gwnnw0J1PUlGDGu1Y84dKtulbdWj
-yrh88D4fSdtmxFbXE7fhYUJTBmEHSUk9OLHh/Tr+nSC3SfH0I/9y6l9j9vVVYhYJ
-C07Z1mZZKVb+gmbbB7LEavGMNaFHjvRJAwBX2TMDbXJceZ9jU/iihILkZbrbG40r
-n1mctYVmcR3YqOzI/ynLje97FEvxtsg99OUjzzXyFMqfAl0J3Gc6tzvAER3N+ovK
-nudsnMB5Y+InQHHmPeizG4mFyeBYesXNwX6cmI30c8KFiAlKHcsxjJsuoBZ3bSwv
-vFdK2hnuCO05HEgCzAQKUlY6Q2F0xJblAgMBAAGjZTBjMA4GA1UdDwEB/wQEAwIF
-oDATBgNVHSUEDDAKBggrBgEFBQcDATAMBgNVHRMBAf8EAjAAMC4GA1UdEQQnMCWC
-C2V4YW1wbGUuY29thwR/AAABhxAAAAAAAAAAAAAAAAAAAAABMA0GCSqGSIb3DQEB
-CwUAA4IBAQBnfO4lYRXR9AdMidpgdITqMEKJik8MvCkpQ+EKQLq3CIGXPt5lkHLs
-ysbF9f3VxioKNYzkakJGVGyu51hqyhGqGQ4M7IpOBQkmY24IExWPVEk2wkIV+HTU
-+oQVZOIrHF+s9IIFOIh3SIPIsXNvx7rUc5sgF4P+eAnAcv3o1zL7YjGJZ8e27Ai2
-uF8iG/po/0Vd93OSB8Tj/Nvg99SSucy7nBYTreSdhUjZWRI0W1oYJX49/fhWljR9
-8+f2GqUfLc7iCjcV3wxlfBqEKCdpjXsiqtsb1KrAx7AEOj7XfDjJjyCL4bshLp9x
-PbV+kBFCN151iWYtfzhKEplrZFYNXlX2
+MIIDOTCCAiGgAwIBAgIQNa9+JgaO+xR8zZNBEpuWFzANBgkqhkiG9w0BAQsFADAS
+MRAwDgYDVQQKEwdBY21lIENvMCAXDTcwMDEwMTAwMDAwMFoYDzIwODQwMTI5MTYw
+MDAwWjASMRAwDgYDVQQKEwdBY21lIENvMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8A
+MIIBCgKCAQEA0m/AL3KP4t8K3izOO9HsoPcJh/Be5qX/oXPk5TZPUfAs7OQDFoFv
+TZ4LZXd46oNTzRRP/Mk5knMrwtkFLWX+eh7AnXmerUheN2zQ58EpgwN42B3tyk3g
+Msa0Rs/mG5gUqkUXAE0aKWqx5RjzDa8nKbxhKTtGJSMoyM3ZobTDhu2aZWXTVWjx
+68Mh/g5LhmPjVewYWu0wOFToHfP5K481CpMSHfpOHRxeXMuw7TWAW7jo7AjWebrc
+1VFaVjRhOwmpjuAOmlEXlNPQfz2cj9+YkUNPSL3hjYFKxWRf22AMmFAHn1fT4jOp
+XCuWnRbimd7St5DXgiWz3UKD5ekvFoklCQIDAQABo4GIMIGFMA4GA1UdDwEB/wQE
+AwICpDATBgNVHSUEDDAKBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MB0GA1Ud
+DgQWBBRPNyX6AR3LoGGTVDy7+mh2QTPsvjAuBgNVHREEJzAlggtleGFtcGxlLmNv
+bYcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAATANBgkqhkiG9w0BAQsFAAOCAQEAkMha
+P9Q59eoqz+lNwlFW5VIVK3aiEG6EA0clIy21qvhSrWEZ5E4bAc0FgYkvpVtnNUQa
+9KtyQJjhqIKIU2hyCyRDbc6yxZiaeKMGamB962+aioLmpPe6Ix/SRddTE9kyAWfc
+MyltZw9wErtTHHk5JOjuBAdsuTMs/AeUIXSgtjoMoWgbD3kgVoW6V84j0IFbJ5RS
+XTSF6CuKwy4MAlgROBjeqzj0dANlJwNadxHNjVtzZ/BsF20AX93dOKZgR9/Cu8Dl
+/ghKA8SiB0Er3RCGLnK8vr4MLt8r7C8SwEDs+ocd/n3Uw+Vv/0slTgXleOos1GpK
+o1i3NwuQ7fa3oKPL4g==
 -----END CERTIFICATE-----`)
 
 // localhostKey is the private key for localhostCert.
 var localhostKey = []byte(testingKey(`
 -----BEGIN RSA TESTING KEY-----
-MIIEvgIBADANBgkqhkiG9w0BAQEFAASCBKgwggSkAgEAAoIBAQDbsEfk1bK7ozwZ
-lcQM8rBUikC4gwnnw0J1PUlGDGu1Y84dKtulbdWjyrh88D4fSdtmxFbXE7fhYUJT
-BmEHSUk9OLHh/Tr+nSC3SfH0I/9y6l9j9vVVYhYJC07Z1mZZKVb+gmbbB7LEavGM
-NaFHjvRJAwBX2TMDbXJceZ9jU/iihILkZbrbG40rn1mctYVmcR3YqOzI/ynLje97
-FEvxtsg99OUjzzXyFMqfAl0J3Gc6tzvAER3N+ovKnudsnMB5Y+InQHHmPeizG4mF
-yeBYesXNwX6cmI30c8KFiAlKHcsxjJsuoBZ3bSwvvFdK2hnuCO05HEgCzAQKUlY6
-Q2F0xJblAgMBAAECggEACzZIOQraBB8M3G5rEtEZBDuJGZGgggpSXDrsQC22mouV
-M6JiEuOT5Xfdagz10rF5h9lp6DCqsA8/bA7ViWJpYT1BQNwkdGWvC4Oz3EaxDRue
-kjLCqyCmKMCBvfbmAtNsC/G6T5/pNQKTQNlk2YrXd1l2nUUpyBlAHq2bX52jwSGD
-bFy5hyzSrzjeLpLUNZ56W/uXCvP0l6PAEvXRn/KG89XLZCtMBvVDMCfjIe77Q1U9
-/XzIrnb67RzQwiDelvX+biMkBrjeYw/Gvdo9hNCOfbOZ+SpnfDOLEfAha/XPmr3+
-5EeF4emeEhCODvfe7wy/4h1gHEG2N435S61DcV3gQQKBgQD92EJidwriPGDTUSM8
-nJrPQ5xwPMKz5hWpfI0zxIYZyqA37eRC5Q9WD3rDbrEZiLCInFh+Ci899iLzEpFZ
-dFQAUiRam+zFpDCQcGHr/uytRoTH/nxh2MrYPq8cA5ZGU6oMH+Yl4TynqJm2KN7e
-0ocE07QjyK/9nIvEtdibEiFEwQKBgQDdjcgoqHaM49YJ4yxGpjuRdc5a3iuKzZU6
-BON4GKqYQ9u/o8/NPaOSQ3vKhwzTjiEoOZImn+eX1cRP0ZskmQ+LyzsdVAHMDydz
-9I23dbIywtCXGhKOJRwt9O++8ataWIxi1frjj6BcI+TzGl8LM2lYIfUHzVzfswwE
-1EK8ikxnJQKBgBqPKvr0a54aJSNXBPHNjOEMuOyBXvnFpBSUpI17DXDbY4IWkOBy
-6PTfL8AM79i1FYtlmFivphu8ihGWqsCKTFOwRH96ev5+3FnweD5h8M98Zl4qgUcX
-kLmpbVboBSwcitkz6TejZl5AZLzLb+4uZtQZdmqcD9XgMDuHrz8iWXrBAoGBAMJO
-Z34pCRfVddFkGF+5yMJw5FLTSLLKTJb+1JRuZad21BIF0+i3p25OmxHrUXd0zmWd
-4CzZzt5eD3bFaOA3EOhUi/rTw2O44qwSjfuZUHiuXQw4RI+/wjAYAe+fud1ZjX3d
-FtVfEI/etxvyQ+rp4vj1hxWZqVtThzXxBrqePBW1AoGBAOTC19rFQXtVf9A+8c/w
-2ryAY2W9qNKe0xMivTAqau0Kdy2/2toJekR/5qOy3tOF7JasOgG+y3m3gLF47EFF
-v75eW4FkiCFvsyl/qv4CO1eKnHlvkRoDsnMb+dA5czst58rO6BK40QvPqwXaSxj1
-ee8ReNCDhC0Zidczajm63O1G
+MIIEvAIBADANBgkqhkiG9w0BAQEFAASCBKYwggSiAgEAAoIBAQDSb8Avco/i3wre
+LM470eyg9wmH8F7mpf+hc+TlNk9R8Czs5AMWgW9Nngtld3jqg1PNFE/8yTmScyvC
+2QUtZf56HsCdeZ6tSF43bNDnwSmDA3jYHe3KTeAyxrRGz+YbmBSqRRcATRoparHl
+GPMNrycpvGEpO0YlIyjIzdmhtMOG7ZplZdNVaPHrwyH+DkuGY+NV7Bha7TA4VOgd
+8/krjzUKkxId+k4dHF5cy7DtNYBbuOjsCNZ5utzVUVpWNGE7CamO4A6aUReU09B/
+PZyP35iRQ09IveGNgUrFZF/bYAyYUAefV9PiM6lcK5adFuKZ3tK3kNeCJbPdQoPl
+6S8WiSUJAgMBAAECggEAH6wNnSaNNOx0domAVb0L9BMSVmktY2IPFjX6+Kuipxay
+sEnGFDRRc/lqRjCD3Kn86iQZQNbpaRsX9kjTEGjmMuufemQUZ4i5k3Z3f7YRISSb
+4s0oXbc4W/QPrIu8WqFaVOE2q8cu/ZZPJYfKhx/6UTwG1J2U+mBvEj/3NjEeb5db
+yPAUu57y2XE2lIGZrjFfhp0b7lP779PVix4zDRv4uim0SifOvEWS2QUkwktWvdi+
+CQ8sExelHmyiOmqYNUYhfyeLiiFmY3CLiuLQlsKCU3fQH5RMY4lSU3WfrWDU/9hu
+Ub6hOwC2o4EGcipOT9NIKTT6xaO5BtkWvXMBowr6vQKBgQDYfKqJY+53Wft7Awsp
+UcmnCOCM/o0hRpCIHD2CEhWp3d3EjmBLkapN37VLUqflHgy1K9XXcj/JsbQebBKX
+EnqrnAttzXDyhK0rHse/h6wipvb5yNeNXuEq+4eqWOn8xAHqZ2oPNYro7k8vg/xs
+O+z0JnZxqtQWCkTHbkENiBQnbQKBgQD42GC4IW+eMzqvQrEC8yBo+v7nX+GUW30V
+cx4FszyEmTCLgmbTZDPAzv9RHxbZJFekT26Mv6KsrCSkoS/9VFO4ScLZ8hM9j5xk
+PClW/h2RBj15IVb5kBhnPpipXP80NIesy7q+fNCtejRBzDqMUgDy5TBu2AoGjuRx
+LnRllGJmjQKBgE+Rwrys1kFX+W7FN50ASrXyNVadWRIZ9ZWRi6n7K6s7uP89tXcR
+Gx90UhZVUxCLcvBEJF0DyR5BCosD/fE5AY2B1VGOaS06RsvKjHCBVWuHIs+8Wj3D
+2c8KlFqwQ5B8GpglWn4sl9n1fXVhda3yeK1kVbSBrWJMc+s5Qv/OPFohAoGAH7tP
+0gsJda3orNKfSdYA6wIeoRh75nQoJiUQzf9GF06FfzS4sfoUE2ER86E4Q6Y7ussa
+OL5zGvu0jH21ZeNr7ZMSUsJoq6/1bF6zktFCqeyWyCx1L8utIPU9eFNQg7KzHli4
+TnQSFB0CLuXBDYWjc1XAcFICLUXpYXlgBFgAmD0CgYBOPw/bINcuQR50FKqGSH8H
+Hg/6eBSf6BVoMQmS9FYKkr/qjYBmXs9O99Ak7HKT0xr56FNTDRtpMkLkJJJcHpif
+4odTUKveh5bpAh3q4mX0j2r3HAo341i8cNQHiuh7YQfqlriDW3ejChx1jvy2GV0G
+jU5Li19dcUhK+0KxTXl9tA==
 -----END RSA TESTING KEY-----`))
 
 func testingKey(s string) string { return strings.ReplaceAll(s, "TESTING KEY", "PRIVATE KEY") }
